### PR TITLE
Add virtual network support to ACI

### PR DESCRIPTION
**Pull request by chelnak**
This PR adds ACI networking support and the ability to whitelist subnets on a posgres server firewall.

**Changes:**
* Updated existing ACI block
* Updated existing postgres server firewall block to handle virtual networks
* Added virtual network block
* Added a network profile block to be used with ACI
* Added an example template that consumes the above blocks

**Example**
The example assumes that all resources are in the same resource group and that a posgres server already exists.

It aims to demonstrate the scenario where there may be a requirement to run an ACI instance that communicates with a postgres server that does not have `allow access to azure services` turned on.

```PowerShell
New-AzResourceGroupDeployment -Name aci-network -ResourceGroupName poc-vnet-integration -TemplateFile .\examples\example-aci-with-networking.json -servicePrefix cg -postgresServerName posgres1
```

### CODE REVIEW

**Review by smneal — APPROVED** [review]
Looks like quite a lot!

**Review by rizzkhan1 — COMMENTED** [review]
nice one!

File: templates/container-instance-network-profile.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    "interfaceIpConfig": {
+      "type": "string",
+      "defaultValue": "ipConfig1",
+      "metadata": {
+        "description": "The name of the ip config for the interface"
+      }
+    },
+    "vnetName": {
+      "type": "string",
+      "defaultValue": "",
```
Comment (EwanNoble) [inline templates/container-instance-network-profile.json]: Probably shouldn't be optional
```suggestion
```

File: templates/container-instance-network-profile.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      "metadata": {
+        "description": "The name of the vNet that the container instance will connect to"
+      }
+    },
+    "subnetName": {
+      "type": "string",
+      "defaultValue": "",
```
Comment (EwanNoble) [inline templates/container-instance-network-profile.json]: Nor this
```suggestion
```

File: templates/container-instance-network-profile.json [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      }
+    }
+  },
+  "resources": [
+    {
+      "name": "[parameters('networkProfileName')]",
+      "type": "Microsoft.Network/networkProfiles",
+      "apiVersion": "2018-07-01",
+      "location": "[resourceGroup().location]",
+      "dependsOn": [],
```
Comment (EwanNoble) [inline templates/container-instance-network-profile.json]: Can remove this while we're at it
```suggestion
```

**Review by EwanNoble — APPROVED** [review]

**Pull request merged**